php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -823,7 +823,7 @@ abstract class NDB_BVL_Instrument extends NDB_Page
         //
         // json_encode ensures that this is safe. If we use the safe wrapper,
         // HTML encoding the quotation marks will make it invalid JSON.
-        $db->unsafeUpdate(
+        $db->update(
```

Comment:
In this case, it is unsafe to use `unsafeUpdate` but it's also problematic to use `update()`. 

with update(), the json daat saved will be `htmlspecialchar`ed and thus will no longer be a valid JSON format, when the `getinstancedata()` function attempts to run decode on it, the decoding will fail.

My proposal:
1. use unsafeupdate to insert the data in the database
2.  modify the loadInstance data to run `HTMLSpecialChars()` on the decoded data only as follows
   ```
        if ($this->jsonData) {
            $jsondata = $db->pselectOne(
                "SELECT Data FROM flag WHERE CommentID=:CID",
                ['CID' => $this->getCommentID()]
            );

            $this->instanceData = json_decode($jsondata, true) ?? [];
            $this->instanceData = array_map(
                'htmlspecialchars',
                json_decode($jsondata, true) ?? []
            );
        } else {
   ```


@driusan thoughts?

Reply:
@driusan you got feedback on this ? security wise

Reply:
@driusan Just wanted to check if this has made it's way to you yet!

Reply:
I'm having a bit of trouble wrapping my head around the proposal but I *think* it makes sense in that the behaviour should be similar to what we have now. My concerns are:

1. What about non-HTML places that use instanceData? (ie. API responses) Maybe we should do it right and just do the `htmlspecialchars` at the template level. Or would that break other things that depend on the current behaviour?
2. It's not really clear to me why we're changing this in this PR at all. What does this have to do with moving data out and into its own column?

Reply:
@driusan 

> What about non-HTML places that use instanceData? (ie. API responses) Maybe we should do it right and just do the htmlspecialchars at the template level. Or would that break other things that depend on the current behaviour?

Good point but it's probably not the concern of this PR

>  It's not really clear to me why we're changing this in this PR at all. What does this have to do with moving data out and into its own column?

Also good point, my guess is we discovered the security flaw while testing this ? I don't see why it;s related.

Reply:
@CamilleBeau i think you should just put it back to unsafe update and before we release we have to address the instrument encoding issue which is LONG overdue